Tree.h

```diff
@@ -132,13 +132,21 @@ class LeafNode : public TreeNode<T> {
 
 // load a regression tree from Json
 template <class T>
-TreeNode<T>* fromJson(const folly::dynamic& obj) {
-  int index = obj["index"].asInt();
+TreeNode<T>* fromJson(const folly::dynamic& obj, const Config& cfg) {
+  const folly::dynamic* feature = nullptr;
+  try {
+    feature = &obj["feature"];
+  } catch(...) {
+  }
```

Comment:
it should work, https://github.com/facebook/folly/blob/master/folly/docs/Dynamic.md

auto pos = obj.find("no_such_key);
    // pos == obj.items().end()

the folly is local built, we built in chef cookbook:

bash 'install_folly' do
  cwd '/mnt/projects/folly/folly/'
  user "root"
  code <<-CMDS                                                                                                                          
     autoreconf -ivf                                                                                                                    
     ./configure --with-boost=/usr/local CPPFLAGS="-I/usr/local/include" LDFLAGS="-L/usr/local/lib" GFLAGS_NAMESPACE=google             
     make && make check && make install                                                                                                 
  CMDS                                                                                                                                  
end